apps/src/lib/levelbuilder/lesson-editor/FindProgrammingExpressionDialog.jsx

```diff
@@ -0,0 +1,217 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+import Button from '@cdo/apps/templates/Button';
+import DialogFooter from '@cdo/apps/templates/teacherDashboard/DialogFooter';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+
+import LessonEditorDialog from './LessonEditorDialog';
+
+const SearchForm = function(props) {
+  return (
+    <form className="form-search">
+      <input
+        type="text"
+        className="input-large search-query"
+        onChange={props.onSearch}
+        placeholder="search by name"
+      />
+      <select
+        className="input-small search-query"
+        style={{marginLeft: 4}}
+        onChange={props.onFilter}
+      >
+        <option value="">filter</option>
+        {/* TODO: these should probably be sourced from somewhere, rather than hardcoded */}
+        <option value="applab">Applab</option>
+        <option value="gamelab">Gamelab</option>
+        <option value="spritelab">Spritelab</option>
+        <option value="weblab">Weblab</option>
+      </select>
+    </form>
+  );
+};
+
+SearchForm.propTypes = {
+  onSearch: PropTypes.func.isRequired,
+  onFilter: PropTypes.func.isRequired
+};
+
+const ProgrammingExpressionTable = function(props) {
+  if (
+    !props.programmingExpressions ||
+    props.programmingExpressions.length === 0
+  ) {
+    return null;
+  }
+
+  // TODO what should we actually do here?
```

Comment:
open to suggestions!

Reply:
this UI accomplishes a lot! If the question is how the user interface should be, I think what you've built is good for launch 🚀

Reply:
oh, to clarify; the question is what we should do when w programming expression does not actually have a color associated with it

Reply:
interesting. perhaps the same styling as well-formed doclinks, but with a foreground/background color combo that is visually distinct? if that's not possible, I'd rather have visually distinct from all other doclinks, rather than picking a default color that might be confused for well-formed doclink.

Reply:
oh, that makes sense; we can just leave out the color, and style them as regular code blocks

Reply:
sounds good to me!